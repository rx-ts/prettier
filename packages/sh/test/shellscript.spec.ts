import prettier from 'prettier'
import ShPlugin from 'prettier-plugin-sh'

test('parsing a simple program', () => {
  const output = prettier.format(`echo      'foo'`, {
    // @ts-ignore
    parser: 'sh-parse',
    plugins: [ShPlugin],
  })

  expect(output).toMatchSnapshot()
})

test('fatal parse error', () => {
  expect(() =>
    prettier.format(`echo )`, {
      filepath: 'broken.sh',
      // @ts-ignore
      parser: 'sh-parse',
      plugins: [ShPlugin],
    }),
  ).toThrow()
})
