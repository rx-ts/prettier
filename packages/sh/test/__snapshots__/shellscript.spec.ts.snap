// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsing a simple program 1`] = `
"echo 'foo'
"
`;
